src-java/nbworker-topology/nbworker-storm-topology/src/test/java/org/openkilda/wfm/topology/nbworker/validators/PathValidatorTest.java

```diff
@@ -496,6 +506,46 @@ public void whenDiverseWith_andExistsIntersection_validatePathReturnsError() {
                         + " port 6 and destination switch 00:00:00:00:00:00:00:03 port 6.");
     }
 
+    @Test
+    void whenDiverseWith_andIntersectionWithGroup_validatePathReturnsAllIntersections() {
+        Switch switch1 = Switch.builder().switchId(SWITCH_ID_1).build();
+        Switch switch2 = Switch.builder().switchId(SWITCH_ID_2).build();
+        Switch switch3 = Switch.builder().switchId(SWITCH_ID_3).build();
```

Comment:
I see some places where these switches are created. Maybe we can make them static?

Reply:
yes, could be, thanks for the suggestion. I'll refactor it